### DISCUSSION TITLE
Homebridge3 on Raspberry Pi

### DISCUSSION THREAD

**Discussion by brainstorm-tech** [homebridge-simplisafe3/homebridge-simplisafe3#383] [General]
Hi all,

I run Homebridge on Linux Raspberry PI.  Since Simplisafe's awful new authentication, I've been out-of-luck.  I cannot find a Raspberry web browser that both runs Homebridge-simplisafe3 and also completes the authentication process - debugger shows "unknown protocol" and "mfa errors" and won't complete the authentication.  Raspberry doesn't offer a full Chrome or Firefox web browser, just a stripped-down Chromium and feature-poor Firefox-ESR.  

Has anybody had any success with this, before I go out and buy a $500 dedicated PC to run this?  Is there ANY way I can use a web browser on another device and copy-and-paste an access key from a non-PI system?  Like any of these?

a.authDomain="auth.simplisafe.com";
a.authClientId="DWkIUe6LC38xLomvfG6LXesCCaKJGl24";
a.apiHost="https://api.simplisafe.com";
a.appHubHost="https://app-hub.prd.aser.simplisafe.com";
a.telegramHost="https://telegram.prd.aser.simplisafe.com";
a.graphQlHost="https://graphql.prd.aser.simplisafe.com/";
a.apiPath="/v1";
a.mediaHost="https://media.simplisafe.com";
a.mediaPath="/v1";
a.mediatorHost="";
a.simpliSafeUrl="https://simplisafe.com";
a.nestAuthUrl="https://home.nest.com/login/oauth2?client_id=55db4458-e3a2-4300-afd7-d1287dbe21b2&state=STATE";
a.nestFirebaseUrl="wss://developer-api.nest.com";
a.augustAuthUrl="https://auth.august.com/authorization?client_id=6c7dda55-11c4-43ea-9d4a-fb95bd17bf55&state=STATE&redirect_uri=REDIRECT_URI";

Thanks,
Ed

**Comment by shamoon** [comment]
Hi, not exactly sure I understand but if you're running homebridge on a Pi you can access that pi as the 'server' from somewhere else on your network e.g. your usual computer with a normal browser and complete the auth from there. But yes, you can copy / paste the URL that it generates into your browser. You can also authenticate from the command line (which will kick you out to the browser, but still).

**Reply by brainstorm-tech** [reply]
Thanks Shamoon. Attempting to run from another browser on the network doesn't seem to work.  The "Launch Simplisafe Login" button fails with:

Failed to launch 
api/plugins/settings-ui/homebridge-simplisafe3/index.html?origin=http%3A%2F%2F192.168.0.170%3A8581&v=1.10.11#auth:1 'com.simplisafe.mobile://auth.simplisafe.com/ios/com.simplisafe.mobile/callback?code=QgDZ-IhoZHkAKjFmHoty9AfY2vxu1AJ2It9bSs1ZOBW4n' because the scheme does not have a registered handler.

**Reply by brainstorm-tech** [reply]
The handler error is with Chrome on MacOS.  For Firefox on MacOS, I get:  Prevented navigation to “https://auth.simplisafe.com/authorize/resume?state=0qv-2oOyTuFVjPyP-0DFHFVbshFAknfX” due to an unknown protocol. [mfa-sms-challenge](https://auth.simplisafe.com/u/mfa-sms-challenge?state=hKFo2SAtSEVTdWNGZkYzQ3pIUF9tYWxIUHVYcGR4M2NGb19IWKFusG1mYS1hdXRoZW50aWNhdGWjdGlk2SAwcXYtMm9PeVR1RlZqUHlQLTBERkhGVmJzaEZBa25mWKNjaWTZIDQyYUJaNWxZclZXMTJqZk91dTNDUVJPaXR3eGc5c041)

thanks, 
Ed

**Reply by shamoon** [reply]
Hmm, haven't seen anyone having this issue before. When you click to show the authentication instructions the Launch button is just a link, so you can e.g. right-click and then copy + paste that URL into a browser window, go through the SimpliSafe login and then the final step can be tricky because certain browsers will just not work and its not obvious, Chrome seems to be the easiest because it will show you the final URL you need, which starts with `com.simplisafe.mobile://` and you take that URL and copy + paste it back into homebridge.

See the full instructions: https://github.com/homebridge-simplisafe3/homebridge-simplisafe3#simplisafe-authentication and a short video of the start of the process typically here: https://github.com/homebridge-simplisafe3/homebridge-simplisafe3/discussions/291#discussioncomment-2101078

**Comment by brainstorm-tech** [comment]
Hi Shamoon, thanks.  I don't think I ever closed the loop on this, but I was eventually able to get Chrome to work.  It's giving a weird error but buried deep is the com.simplisafe.mobile:// code which worked.

Thanks for your patience and for this awesome plugin!
-Ed
